### PR TITLE
Release `1.0.0-pre4`

### PR DESCRIPTION
**Pull request by dmitrykuzmin**
This PR releases `1.0.0-pre4` version of this library.

The code base is brought into correspondence with the latest Spine as well as the new [event subscriptions](https://github.com/SpineEventEngine/core-java/pull/961) functionality.

Gradle version is updated to 5.2.

### CODE REVIEW

**Comment by dmitrykuzmin** [comment]
@armiol PTAL.

**Review by armiol — COMMENTED** [review]
@dmitrykuzmin can you please apply PMD rules and address the issues?
(see https://github.com/SpineEventEngine/base/pull/325/commits/7c89593e251570288c16293b5288d427a813e7c4)

File: firebase-mirror/src/main/java/io/spine/server/firebase/FirestoreSubscriptionPublisher.java [hunk 1]
```diff
@@ -23,34 +23,28 @@
...
  * href="https://firebase.google.com/docs/firestore/">Cloud Firestore^</a>.
+ *
+ * <p>The descendants elaborate on how to compose the document identifier and body for this
+ * particular subscription.
+ *
+ * @param <U>
+ *         the subscription update payload format
  */
-final class FirestoreSubscriptionPublisher {
+abstract class FirestoreSubscriptionPublisher<U> implements Logging {
```
Comment (armiol) [inline firebase-mirror/src/main/java/io/spine/server/firebase/FirestoreSubscriptionPublisher.java]: Isn't it just `FirestorePublisher`? And `U` seems to be just an "update" we publish.

File: firebase-mirror/src/main/java/io/spine/server/firebase/EventSubscriptionPublisher.java [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+
+/**
+ * A subscription update publisher for the event subscriptions.
+ *
+ * <p>The document identifier is an event ID and the composed document body contains the serialized
+ * event message as well as some meta-information.
+ *
+ * <p>See {@link EventStateField} for the details on storage format.
+ */
+final class EventSubscriptionPublisher extends FirestoreSubscriptionPublisher<Event> {
```
Comment (armiol) [inline firebase-mirror/src/main/java/io/spine/server/firebase/EventSubscriptionPublisher.java]: If `FirestoreSubscriptionPublisher` is just `FirestorePublisher` (see my comment below), this is `EventPublisher`.

File: firebase-mirror/src/main/java/io/spine/server/firebase/EventUpdateObserver.java [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+import java.util.List;
+
+/**
+ * A {@code SubscriptionUpdateObserver} for the event subscriptions.
+ *
+ * <p>Publishes a new event which occurred in the system to the underlying Firestore.
+ *
+ * @see EventSubscriptionPublisher
+ */
+final class EventUpdateObserver extends SubscriptionUpdateObserver<Event> {
```
Comment (armiol) [inline firebase-mirror/src/main/java/io/spine/server/firebase/EventUpdateObserver.java]: And this is `EventObserver`, which extends `UpdateObserver`.

**Review by armiol — CHANGES_REQUESTED** [review]
@dmitrykuzmin and also please see my comments.

**Comment by dmitrykuzmin** [comment]
@armiol PTAL.

**Review by armiol — APPROVED** [review]

**Pull request merged**